### PR TITLE
Simplify owners

### PR DESCRIPTION
**Pull request by pauljohanneskraft**
# Simplify Owners

## :recycle: Current situation & Problem
Storing ownership of an organization inside the organization makes is quite complex to check for authentication and to store users in Firestore. By making ownership another UserType, we simplify user creation, invitation logic and security rules.

## :gear: Release Notes 
*Add a bullet point list summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented or links to the documentation if it appends or changes the public interface.*

## :books: Documentation
*Please ensure that you properly document any additions in conformance to [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).*
*You can use this section to describe your solution, but we encourage contributors to document your reasoning and changes using in-line documentation.* 

## :white_check_mark: Testing
*Please ensure that the PR meets the testing requirements set by CodeCov and that new functionality is appropriately tested.*
*This section describes important information about the tests and why some elements might not be testable.*

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,7 +134,7 @@ In this section, we describe all user-related data to be stored. The security ru
 
 |Property|Type|Values|Comments|
 |-|-|-|-|
-|type|optional string|e.g. "admin", "clinician", "patient"|The type of the user. It may be set to null for owners or users not requiring any further permissions.|
+|type|string|e.g. "admin", "owners", "clinician", "patient"|The type of the user.|
 |dateOfEnrollment|Date|-|The date when the invitation code was used to create this user.|
 |dateOfBirth|optional Date|-|The date when the user was born.|
 |genderIdentity|optional string|-|The gender identity chosen when a patient redeemed the invitation. TBD: Available values.|
```
Comment (PSchmiedmayer) [inline README.md:140]: Would be great to address the TBD here.

File: firestore.rules [hunk 2]
```diff
@@ -79,71 +71,52 @@ service cloud.firestore {
...
+      allow read: if isAdmin() 
+        || isUser(userId) 
+        || isOwnerOrClinicianOfSameOrganization();
 
-      allow read: if isAuthenticated() && (isUser(userId) || isClinicianReadable());
-      allow write: if isAuthenticated() && (
-        (isUser(userId) && isPatientWritableCollectionName())
-          || isClinicianWritable()
-      );
+      allow write: if isAdmin()
```
Comment (PSchmiedmayer) [inline firestore.rules:117]: Not sure if I am missing this, but we need to ensure that the organization that is set here is only set by someone that is part of the organization (e.g. You are an owner of the Stanford Org. Same for me. As an owner of the Stanford org I shouldn't be able to add JH to your org list and make you an owner on this org).

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the change; only had a small comment about the security rule.

File: firestore.rules (hunk 2 above)
Comment (pauljohanneskraft) [reply firestore.rules:117]: This is the rule for user collections (e.g. `users/{userId}/bodyWeightObservations`), not for the user itself - the user has its own match statement above with differing rules

File: README.md (hunk 1 above)
Comment (pauljohanneskraft) [reply README.md]: done

**Pull request merged**